From ca47018d78159168a1eae12b1871d879cddcc9ba Mon Sep 17 00:00:00 2001
From: Hans de Goede <hdegoede@redhat.com>
Date: Wed, 1 Feb 2023 16:14:08 +0100
Subject: [PATCH 7/7] Don't rename the already registered PCI-device

devices must not be renamed after they have already been
added to the device hierarchy.

dev_set_name() MUST only be called before dev_add()
and the pci_dev passed to ipu_pci_probe() has been
added long before ipu_pci_probe() runs, so it must
not rename it.

Renaming it is confusing udevd which now all of a sudden sees
a device it already knows about change name.

This is causing udev rules to not work properly with
the ipu6-driver. Drop the clearly wrong dev_set_name() call.

Signed-off-by: Hans de Goede <hdegoede@redhat.com>
---
 drivers/media/pci/intel/ipu.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/drivers/media/pci/intel/ipu.c b/drivers/media/pci/intel/ipu.c
index f1a0be6d9..80ba8fb36 100644
--- a/drivers/media/pci/intel/ipu.c
+++ b/drivers/media/pci/intel/ipu.c
@@ -432,7 +432,6 @@ static int ipu_pci_probe(struct pci_dev *pdev, const struct pci_device_id *id)
 	if (!isp)
 		return -ENOMEM;
 
-	dev_set_name(&pdev->dev, "intel-ipu");
 	isp->pdev = pdev;
 	INIT_LIST_HEAD(&isp->devices);
 
-- 
2.39.1

